Enhance/tests

deivid-rodriguez opened the pull request:
#### :tophat: What? Why?

Improves our test suite:
* Fix duplicated or bad test descriptions.
* Inline some shared specs that were not really shared.
* Minor refactorings.
* Explicitly set controller under test for controller tests instead of some magic that was prone to errors and not really worth it. 

#### :pushpin: Related Issues
_None_.

#### :clipboard: Subtasks
_None_.

### :camera: Screenshots (optional)
_None_.

#### :ghost: GIF
![cirque](https://user-images.githubusercontent.com/2887858/28568672-e80d9d4a-7136-11e7-8b3a-15cb7fddadda.gif)

beagleknight commented:
I am gonna create the `v0.4.3` version now. If you don't mind I will include this PR in the next release 😄 .

mrcasals commented on decidim-core/spec/shared/with_promoted_participatory_processes_examples.rb:
```diff
@@ -53,6 +53,7 @@
 
       expect(controller.helpers.promoted_participatory_processes.count).to eq(3)
       expect(controller.helpers.promoted_participatory_processes).not_to include(unpublished)
+      expect(controller.helpers.promoted_participatory_processes).not_to include(unpromoted)
```
I know I'm late, but I usually do `expect(array).to match_array([first, second, last])`.

The `match_array` method checks that the values of the arrays are the same, without considering the order. This way I don't need to specifically check that some values do not appear in the checked array.

deivid-rodriguez replied:
Yup, I agree with you! I just used this because the other `include` was already there, but I can change it!

deivid-rodriguez replied:
Updated via 143bed2. The best use case of `match_array` I've ever seen :laughing:.

deivid-rodriguez commented:
@mrcasals I changed to your suggested style in ff0adad, it looks much better. I force-pushed onto the commit you commented so I think I deleted your comment too... sorry :sweat_smile:

mrcasals commented:
No problem at all! :)
